kafka/src/main/java/io/micronaut/configuration/kafka/seek/DefaultKafkaSeeker.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.configuration.kafka.seek;
+
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.annotation.NonNull;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.OffsetAndTimestamp;
+import org.apache.kafka.common.TopicPartition;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import static java.util.Collections.singletonList;
+import static java.util.Collections.singletonMap;
+
+/**
+ * Default implementation of {@link KafkaSeeker}.
+ *
+ * @param consumer the consumer on which to perform the {@link KafkaSeekOperations seek operations}.
+ * @author Guillermo Calvo
+ * @see KafkaSeeker
+ * @since 4.1
+ */
+@Internal
+record DefaultKafkaSeeker(@NonNull Consumer<?, ?> consumer) implements KafkaSeeker {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DefaultKafkaSeeker.class);
+
+    /**
+     * Creates a new instance.
+     *
+     * @param consumer the consumer on which to perform the {@link KafkaSeekOperations seek operations}.
+     */
+    public DefaultKafkaSeeker {
+        Objects.requireNonNull(consumer, "consumer");
+    }
+
+    @Override
+    public boolean perform(@NonNull KafkaSeekOperation operation) {
+        try {
+            return Optional.of(operation).filter(op -> op.offset() == 0L).flatMap(this::optimized)
```

Comment:
About the style of code: I'm a bit confused why would you create two methods especially with optional and use flatMap etc. I would simply inline both of them

Reply:
I had decomposed it into two smaller methods for readability. But if it's confusing, I can live with a chunky method.

I inlined those methods here: c24328132d3d150e1fe8c40ec9515ae38f1eb05b